### PR TITLE
Elaborate on build requirements

### PR DESCRIPTION
**Pull request by mikini**
* add "Build system" section detailing generic build requirements;
  * gcc
  * golang
  * ninja
  * cmake
* refer to above section in "Build for Linux"
* use apt instead of apt-get in installation instructions

### CODE REVIEW

**Comment by kasperl** [comment]
Thank you! This is the first external contribution to the Toit language implementation, so give us a few hours an we will have the CLA process fully worked out :)

**Comment by kasperl** [comment]
The CLA process has been documented here: https://github.com/toitlang/toit/blob/master/CONTRIBUTING.md.

**Comment by kasperl** [comment]
Let me know if you need additional information from us, @mikini!

**Comment by kasperl** [comment]
First test of the CLA bot integration: @cla-bot check.

**Comment by mikini** [comment]
Hi Kasper, nice to be able to kick-start the work on establishing a contribution procedure ;).

Sorry for the non-responsiveness the last days, been rather busy. I am quite meticulous regarding rights and contracts, but spent some time reviewing the CLA this evening.

In any case, I think you might ease potential contributors' review of the Toitware CLA by being frank about it being a derivative of the [Apache ICLA](https://www.apache.org/licenses/contributor-agreements.html#clas) (Individual Contributor License Agreement ([full text as pdf](https://www.apache.org/licenses/icla.pdf)). A [diff between Toitware CLA and this](https://paste.sr.ht/~mikini/e3b0f2362e249938ad6e5cf726e987ce3c8a9239) reveal mostly Apache->Toitware text replacements and removal of some of the practical instructions regarding signature (diff formatted in [word-diff plain mode](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---word-diffltmodegt)). Having a reference to this well-known CLA makes it a lot easier to find advise and positions about the pros and cons of signing contracts such as the Toitware CLA.

When you require contributors to sign this CLA, you ask them by paragraph 2 to grant Toitware extensive rights to their contributions which in practical terms are equivalent to reassigning copyright altogether as it includes the right to re-license under a license of Toitware's future choice (quote from ["Apache License Yes, Apache CLA No" article](https://meshedinsights.com/2016/01/07/apache-license-yes-apache-cla-no/): "broad copyright licenses with no limits on use or relicensing. All the same, they convey rights as broad as a copyright assignment would do").

This spawns some questions in my mind, and some doubts about the motivations of open sourcing Toit. I think you need to elaborate some more about how you, both in more abstract philosophical, business and practical terms, see the relationship between Toitware/Toit and the community. If you want to instigate, and benefit from, a healthy community around Toit I think it is crucial to have these important details explicitly stated and be as transparent as possible about it (quote from ["CLA proliferation and the Island of Dr. Moreau" article](https://opensource.com/article/19/1/cla-proliferation): "This is not a happy path and is highly erosive to the open source/collaborative development model").

A couple of the main questions for starters;
* Do you plan to include the open source Toit in closed source derivative deliveries from Toitware?
    * Plain contributions, without CLA, would always be LGPL, but you require the CLA, so this is apparently not regarded as adequate.
* Did you consider the less administrative burdensome and lower barrier-of-entry DCO (Developer Certificate of Origin) route instead of requiring an up-front CLA? ([see article "CLA vs. DCO: What's the difference?"](https://opensource.com/article/18/3/cla-vs-dco-whats-difference) and [wikipedia DCO article](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin))
    * If so, what was the reasons for it not being adequate?

Hope this is not a too inconvenient level of detail, but I think this is an important discussion to have up-front.

**Comment by kasperl** [comment]
Hi @mikini! Thank you for your detailed comments. I will update the documentation around contributions to clarify and hopefully answer your questions.

**Comment by kasperl** [comment]
Landed the clarifying guidelines for contributors; see [CONTRIBUTING.md](https://github.com/toitlang/toit/blob/master/CONTRIBUTING.md). Let me know if this answers your questions, @mikini! I'm not going to claim that I believe the explanations will alleviate your concerns, but I hope the transparency is useful.

**Comment by mikini** [comment]
Hi @kasperl, that indeed makes the Toitware position a lot clearer. Thanks.

I'm not exactly jumping up and down in joy to be contributing to a project where the code is spun off in the dark, and where the driving party can potentially take the project out of the open. But the fact that its LGPL/MIT parts will remain available is of course a huge factor for third parties putting effort into Toit.

I'm willing to sign the CLA with this in mind for the trivial changes I've done until now, also to become further enlightened about Toit and its internals. But I'll definitely be thinking about this again if/when I end up wanting to contribute anything substantial.

Still, despite all this legal mumbo-jumbo, a huge thanks for bringing such an exciting technology to the table ;).

**Comment by kasperl** [comment]
@cla-bot check

**Comment by kasperl** [comment]
Re-checking CLA status: @cla-bot check.

File: README.md [hunk 1]
```diff
@@ -45,6 +45,31 @@ in those directories.  These subdirectories are:
...
+* [go >= v1.16](https://go.dev/),
+
+If you are using a Linux distribution with `apt` capabilities, you can
+issue the following command to install these:
+
+``` sh
+sudo apt install build-essential cmake ninja-build golang
+```
+
+Note that many distributions doesn't provide go v1.16 out of the box at the
```
Comment (floitsch) [inline README.md:64]:
```suggestion
Note that many distributions don't provide go v1.16 out of the box at the
```

**Review by floitsch — APPROVED** [review]
LGTM.

**Comment by floitsch** [comment]
Before I commit: you mention that a very recent version of Go is required.
That shouldn't be the case, as some of our build-bots use older versions. Is it because of https://github.com/toitlang/toit/pull/21 ?

**Comment by kasperl** [comment]
Re-checking CLA status: @cla-bot check.

**Comment by mikini** [comment]
Hi Florian. Yes, the 1.16 mentioned was before I developed the fix of #21 and in doubt of it being acceptable for a merge (I'm not that familiar with go nor go-module).

After the merge, the docs ought to say something else. Maybe just leave out a version requirement if the lower barrier using "go get" is not of practical importance to any potential users. If you know any such your are welcome to bump the number and merge. Else I could just remove it altogether.

Speaking of this, are there known requirements for ninja or GCC we should add?

**Comment by floitsch** [comment]
I'm sure there are requirements on Go, ninja and GCC, but I'm unfortunately not aware of them. "Decently recent" is all I could say now.
As such I would just drop it, and add it back when we hear from users that have difficulties.

File: README.md [hunk 2]
```diff
@@ -92,12 +112,12 @@ You can install this support on most Linux distributions by installing the `gcc-
 use `apt-get`, you can use the following command:
```
Comment (kasperl) [inline README.md:112]: Change this to 'apt' too?

File: README.md [hunk 3]
```diff
@@ -45,9 +45,29 @@ in those directories.  These subdirectories are:
...
 
+### Build system
+
+To build Toit and its dependencies the build host requires;
+
+* [GNU Make](https://www.gnu.org/software/make/)
+* [CMake >= 3.13.3](https://cmake.org/)
+* [Ninja](https://ninja-build.org/)
+* [GCC](https://gcc.gnu.org/)
+* [Go](https://go.dev/),
```
Comment (kasperl) [inline README.md:56]: Drop the comma at the end (for consistency with the other lists).

File: README.md [hunk 4]
```diff
@@ -45,9 +45,29 @@ in those directories.  These subdirectories are:
 
 ## Dependencies
 
+### Build system
+
+To build Toit and its dependencies the build host requires;
```
Comment (kasperl) [inline README.md:50]: Use colon instead of semicolon (for consistency with the other uses of colon).

**Review by kasperl — APPROVED** [review]
Looks good to me. Thank you!

File: README.md (hunk 3 above)
Comment (mikini) [reply README.md:56]: Ahh sorry, had a rebase with this fixed which went stale while waiting for CLA stuff to resolve.

**Pull request merged**